src/stores/Store.ts

```diff
@@ -74,11 +86,268 @@ export default class Store {
 
 	@observable isSchemaApplied = false;
 
+	@observable parsedMessagesHistory: ParsedMessageItem[] = [];
+
+	@observable actMessagesHistory: ActMessageItem[] = [];
+
+	@observable indicators: Indicator[] = [];
+
+	@observable editorCode = '{}';
+
+	@observable editMessageMode = false;
+
+	@observable editedMessageIndex = -1;
+
+	@observable editedMessageSendDelay = 0;
+
+	@observable messageListPanelArea = 50;
+
+	@action clearIndicators = () => {
+		for (let i = 0; i < this.indicators.length; i++) {
+			this.indicators[i] = 'indicator-unvisible';
+		}
+	};
+
+	@action reorderMessagesArray = (
+		destinationIndex: number,
+		sourceIndex: number,
+		draggableMessage: ParsedMessageItem | ActMessageItem,
+	) => {
+		if (isParsedMessageItem(draggableMessage)) {
+			const parsedMessageHistoryCopy = this.parsedMessagesHistory.slice();
+			parsedMessageHistoryCopy.splice(sourceIndex, 1);
+			parsedMessageHistoryCopy.splice(destinationIndex, 0, draggableMessage);
+			this.parsedMessagesHistory = parsedMessageHistoryCopy;
+		} else if (isActMessageItem(draggableMessage)) {
+			const actMessagesHistoryCopy = this.actMessagesHistory.slice();
+			actMessagesHistoryCopy.splice(sourceIndex, 1);
+			actMessagesHistoryCopy.splice(destinationIndex, 0, draggableMessage);
+			this.actMessagesHistory = actMessagesHistoryCopy;
+		}
+	};
+
+	@action setSelectedMethod = (methodName: string | null) => {
+		if (this.selectedService) {
+			this.getServiceDetails(this.selectedService).then(() => {
+				this.selectedMethod = this.serviceDetails?.methods.find(method => method.methodName === methodName)
+					|| null;
+			});
+		}
+	};
+
+	@action setEditedMessageSendDelay = (delay: number) => {
```

Comment:
store is getting really big, let's move all message-list related data and methods into new store